src/app/modules/submission/multirespondent-submission/multirespondent-submission.middleware.ts

```diff
@@ -398,27 +463,27 @@ export const validateMultirespondentSubmission = async (
 
   return (
     // Step 0: Prepare by retrieving relevant reference data
-    okAsync(submissionId)
-      .andThen((submissionId) =>
+    ok(mrfSubmission)
+      .andThen((mrfSubmission) =>
         // Step 0a: If its an existing submission, use the reference data from
         // the submission rather than the form
-        submissionId
-          ? getMultirespondentSubmission(submissionId).map((submission) => ({
+        mrfSubmission
+          ? ok({
               previousSubmission: {
-                encryptedContent: submission.encryptedContent,
-                version: submission.version,
+                encryptedContent: mrfSubmission.encryptedContent,
+                version: mrfSubmission.version,
               },
-              workflowStep: submission.workflowStep + 1,
-              workflow: submission.workflow,
-              form_fields: submission.form_fields,
-              form_logics: submission.form_logics,
-            }))
-          : okAsync({
+              workflowStep: mrfSubmission.workflowStep + 1,
+              workflow: mrfSubmission.workflow,
+              form_fields: mrfSubmission.form_fields,
+              form_logics: mrfSubmission.form_logics,
+            })
+          : ok({
```

Comment:
we've hoisted the snapshot retrieval to an earlier step, which is why we now reference the submission itself instead of the submissionId